pages/search.tsx

```diff
@@ -0,0 +1,234 @@
+import { Locale } from 'next-drupal'
+import getConfig from 'next/config'
+import { serverSideTranslations } from 'next-i18next/serverSideTranslations'
+import parse from 'html-react-parser'
+import { Layout } from '@/components/layout/Layout';
+import { Button, Container, Pagination, SearchInput, RoundedTag } from 'hds-react';
+import { useTranslation } from 'next-i18next';
+import Head from 'next/head';
+import { GetStaticPropsContext, GetStaticPropsResult } from 'next'
+import { NavProps, FooterProps } from '@/lib/types'
+import getMenu from '@/lib/get-menu'
+import { useEffect, useState } from 'react';
+import { getSearch } from '@/lib/client-api';
+import { integer } from '@elastic/elasticsearch/lib/api/types';
+import { useRouter } from 'next/router';
+
+type SearchInputValue = string | undefined
+
+interface SearchPageProps {
+  nav: NavProps
+  footer: FooterProps
+}
+
+interface QueryParams {
+  q: SearchInputValue
+}
+
+export async function getStaticProps(context: GetStaticPropsContext): Promise<GetStaticPropsResult<SearchPageProps>> {
+  const { locale, defaultLocale } = context as { locale: Locale, defaultLocale: Locale }
+  const { REVALIDATE_TIME } = getConfig().serverRuntimeConfig  
+  const langLinks = { fi: '/search', en: '/en/search', sv: '/sv/search'}
+  const { tree: menu } = await getMenu('main', locale, defaultLocale)
+  const { tree: themes } = await getMenu('additional-languages', locale, defaultLocale)
+  const { tree: footerNav } = await getMenu('footer', locale, defaultLocale)
+
+  return {
+    props: {
+      nav: {
+        locale,
+        menu,
+        themes,
+        langLinks,
+      },
+      footer: {
+        locale,
+        footerNav,
+      },
+      ...(await serverSideTranslations(locale, ['common'])),
+    },
+    revalidate: REVALIDATE_TIME
+  }
+}
+
+const ItemBlock = (props: any): JSX.Element => {
+  const content = props.result
+  const title = content.title[0]
+  const text = content.field_lead_in === undefined 
```

Comment:
It's not so easy to understand how the text variable value is get

Reply:
Updated code to more readable.